pvlib/ivtools/sdm.py

```diff
@@ -1252,3 +1255,94 @@ def _calc_theta_phi_exact(vmp, imp, iph, io, rs, rsh, nnsvth):
     theta = np.transpose(theta)
 
     return theta, phi
+
+
+def pvsyst_temperature_coeff(alpha_sc, gamma_ref, mu_gamma, I_L_ref, I_o_ref,
+                             R_sh_ref, R_sh_0, R_s, cells_in_series,
+                             R_sh_exp=5.5, EgRef=1.121, irrad_ref=1000,
+                             temp_ref=25):
+    r"""
+    Calculates the temperature coefficient of power for a pvsyst single
+    diode model.
+
+    The temperature coefficient is determined as the numerical derivative
+    :math:`\frac{dP}{dT}` at the maximum power point at reference conditions
+    [1]_.
+
+    Parameters
+    ----------
+    alpha_sc : float
+        The short-circuit current temperature coefficient of the module. [A/C]
+
+    gamma_ref : float
+        The diode ideality factor. [unitless]
+
+    mu_gamma : float
+        The temperature coefficient for the diode ideality factor. [1/K]
+
+    I_L_ref : float
+        The light-generated current (or photocurrent) at reference conditions.
+        [A]
+
+    I_o_ref : float
+        The dark or diode reverse saturation current at reference conditions.
+        [A]
+
+    R_sh_ref : float
+        The shunt resistance at reference conditions. [ohm]
+
+    R_sh_0 : float
+        The shunt resistance at zero irradiance conditions. [ohm]
+
+    R_s : float
+        The series resistance at reference conditions. [ohm]
+
+    cells_in_series : int
+        The number of cells connected in series.
+
+    R_sh_exp : float, default 5.5
+        The exponent in the equation for shunt resistance. [unitless]
+
+    EgRef : float, default 1.121
+        The energy bandgap of the module's cells at reference temperature.
+        Default of 1.121 eV is for crystalline silicon. Must be positive. [eV]
+
+    irrad_ref : float, default 1000
+        Reference irradiance. [W/m^2].
+
+    temp_ref : float, default 25
+        Reference cell temperature. [C]
+
+
+    Returns
+    -------
+    gamma_pmp : float
+        Temperature coefficient of power at maximum power point at reference
+        conditions. [%/C]
+
+    p_mp_ref : float
```

Comment:
A little strange to see `gamma_pmp` (p m p) and then `p_mp_ref` (p underscore m p)

Reply:
Also a little strange to see pmp returned by a function that is ostensibly for finding temperature coefficient. I might argue that it would make more sense to either just return `gamma_pmp` or change the name of the function and also return vmp and imp. But I don't feel strongly about it.

Reply:
Pvsyst doesn't name this parameter, so I'll call it `gamma_pdc` with units of 1/C, same as is used by the PVWatts model. If the function is returning a value with units of 1/C, then `p_mp_ref` isn't needed (it is if the unit is %/C).